### PR TITLE
Update Processing to 4.1.2

### PR DESCRIPTION
**Pull request by Empyreans**
I merged the current `master` into `processing4` branch, updated the dependencies in `project.clj` according to my solution at [#333](https://github.com/quil/quil/issues/333) and updated the Readme and Changelog.
`lein test` runs without a problem.
We still need to deploy the clojars and update the Wiki. I would like to contribute, and wrote you an e-mail.

### CODE REVIEW

**Comment by Empyreans** [comment]
ClojureScript works too, but I think it might need some more work so I'll break it off into a separate branch. I will exclude Commit 968df35 and update p5js separately.

File: RELEASE-NOTES.md [hunk 1]
```diff
@@ -1,3 +1,12 @@
+## 4.0.0
+__11th February 2023_
+
+* updated dependencies on Processing to 4.1.2
```
Comment (nbeloglazov) [inline RELEASE-NOTES.md:4]: Let's mention that p5.js dependency updated to 1.4.0 as well.

File: project.clj [hunk 2]
```diff
@@ -9,14 +9,13 @@
...
-                 [quil/processing-svg "3.5.3"]
-                 [quil/jogl-all-fat "2.4.0-RC"]
-                 [quil/gluegen-rt-fat "2.4.0-RC"]
-                 [cljsjs/p5 "0.9.0-0"]
+  :dependencies [[org.clojure/clojure "1.11.1"]
+                 [quil/processing-core "4.1.2"]
+                 [quil/processing-pdf "4.1.2"]
+                 [quil/processing-dxf "4.1.2"]
+                 [quil/processing-svg "4.1.2"]
+                 [quil/jogamp-fat "2.4.0"]
```
Comment (nbeloglazov) [inline project.clj:17]: I'll need to upload that jar as well, right?

**Review by nbeloglazov — CHANGES_REQUESTED** [review]

**Comment by nbeloglazov** [comment]
Huge thanks for looking into it! 
Ah, just now saw your comment about excluding p5.js update. Can you update this thread once you undo p5.js change?

File: project.clj (hunk 2 above)
Comment (Empyreans) [reply project.clj:17]: Technically it *should* also be possible to use the jar from the official repository directly https://jogamp.org/deployment/v2.4.0/fat/ but I think maven repositories need a certain structure which this path does not provide.
So yeah, for now I would upload jogamp-fat manually :)

**Comment by Empyreans** [comment]
I have removed the commit updating p5!

**Review by nbeloglazov — APPROVED** [review]

**Comment by jackrusher** [comment]
Great to see this merged! Should we maybe push a snapshot to clojars to get people testing it?

**Pull request merged**